File: cpp/c-plus-plus-sum-digits-string/example2.cpp
// https://codevscolor.com/c-plus-plus-sum-digits-string

#include <iostream>

using namespace std;

int findSum(string str)
{
    int sum = 0;

    for (char ch : str)
    {
        if (ch >= '0' && ch <= '9')
        {
            sum += ch - '0';
        }
    }
    return sum;
}

int main()
{
    string str;
    cout << "Enter a string:" << endl;

    getline(cin, str);
    cout << "Sum: " << findSum(str) << endl;
}

File: javascript/javascript-add-digits-number/example2.js
// https://codevscolor.com/javascript-add-digits-number

let n = 679;
let sum = 0;

for(let no = n; no > 0; no = Math.floor(no/10)){
    sum += no % 10;
}

console.log(`Sum of digits of ${n}: ${sum}`);